Code Challenge 14 - PWDHASHING

clamytoe opened the pull request:
My take on playing around with decorators. Was loads of fun and learned a lot in the process. Can't seem to figure out how to submit only my file though. This is my first PR so I hope that I'm doing it right...

pybites commented:
Great to hear. I manually merged your PR, naming the file with your GH user. I added a requirements file and placed passlib on its own line to make it clear it is an external package (I had to pip install it). It is on our community branch now. Thanks, nice examples!

pybites commented:
See here: https://github.com/pybites/challenges/tree/community/14 

How would submitting be easier for you? 
We rather stick with 3 branches: master for new challenges, solutions for our solution, community for submitted PRs. 

Any feedback is welcome though. Thanks

clamytoe commented:
The setup you have now works great and makes the most sense, I'd stick with
that. Thanks for correcting my submission, I had no clue and the changes
you made all make perfect sense. Thanks for the great learning experience
and for hosting these challenges.

On Fri, Apr 14, 2017 at 6:08 PM, PyBites <notifications@github.com> wrote:

> See here: https://github.com/pybites/challenges/tree/community/14
>
> How would submitting be easier for you?
> We rather stick with 3 branches: master for new challenges, solutions for
> our solution, community for submitted PRs.
>
> Any feedback is welcome though. Thanks
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pybites/challenges/pull/9#issuecomment-294253262>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGztD0mezfti7tzThB5gYbU75nuIsxtYks5rv_xigaJpZM4M83Ho>
> .
>



-- 
"I have always wanted to be somebody, but I see now I should have been more
specific."

bbelderbos commented:
Hey Martin, we appreciate your feedback, thanks. I also featured your submission in our review post, great work :)

As with all things it took some hard work upfront but the challenges are getting traction, nice to see. We definitely are learning a lot. 

Next week we do some Flask programming. If you have a nice challenge - something you struggle with and/or want to learn and/or could serve our community - please let us know.

Cheers

clamytoe commented:
Thanks for the mention!

I made a slight change to boxit and renamed the file. It's showing that the
whole file has changed, I think that I probably needed to fork your
community branch and modified it instead of from my files to prevent that.
Sorry, still learning.

This past week I've been working on a sort of command line utility to track
a sort of todo/task list so that I can have a better grasp of my
accomplishments for the year for when it comes time for my "self" review. I
looked at a lot of other todo utilities but they were either confusing,
without documentation, didn't work, or had too many dependencies. Mine uses
a sqlite3 database and argparse to keep track of them. Not sure if you want
to do a challenge on something like that..

On Sat, Apr 15, 2017 at 1:41 AM, Bob Belderbos <notifications@github.com>
wrote:

> Hey Martin, we appreciate your feedback, thanks. I also featured your
> submission in our review post, great work :)
>
> As with all things it took some hard work upfront but the challenges are
> getting traction, nice to see. We definitely are learning a lot.
>
> Next week we do some Flask programming. If you have a nice challenge -
> something you struggle with and/or want to learn and/or could serve our
> community - please let us know.
>
> Cheers
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pybites/challenges/pull/9#issuecomment-294276057>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGztD_wEYlQ6hSRtKwMl4BlOBxsKZLzzks5rwGadgaJpZM4M83Ho>
> .
>



-- 
"I have always wanted to be somebody, but I see now I should have been more
specific."

bbelderbos commented:
OK merged it. Lets try merging it on community branch next time and see if I can auto-merge from the GUI = easier. Thanks. BTW wouldn't you get the community branch if you clone our repo as per INSTALL guide?

I like you keep track of your accomplishments, powerful technique. For our Flask challenge we were thinking about a simple TODO app and/or birthday tracker. Could easily present them both. To focus on getting starting with Flask I rather keep them very simple. Send us an email if you have any specific input and/or for future challenges: pybitesblog@gmail.com - thanks.

clamytoe commented:
I see what happened. Even though I was telling it to push to the community
branch it seemed to only be pushing to the master one because I started my
code in my local master and not in my community one. My community branch
still doesn't even have the 14 folder in it...

> --
"I have always wanted to be somebody, but I see now I should have been more
specific."

pybites commented:
Weird, I tend to merge master into community (before pushing) upon each challenge. Lets try again next time you want to submit something. Cheers

clamytoe commented:
I do have one question.

I went back and started from the first challenge. I purposely have not look
at any of the solutions for it to challenge myself. If I merge that into my
main, and then later on I try to do a PR for any future challenges, how I
do prevent this older challenge from being submitted to your repo? Is that
where the .gitignore file comes into play?

I just started reading GitHub Essentials from packt.pub so hopefully I'll
quit being such a nub. <-- worse then a noob...

Thanks for being such a great help btw.

On Sat, Apr 15, 2017 at 2:24 PM, PyBites <notifications@github.com> wrote:

> Weird, I tend to merge master into community (before pushing) upon each
> challenge. Lets try again next time you want to submit something. Cheers
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pybites/challenges/pull/9#issuecomment-294313015>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AGztD0c9ZysPQ5pUVwm_-8lwFeHDwXQ_ks5rwRl-gaJpZM4M83Ho>
> .
>



-- 
"I have always wanted to be somebody, but I see now I should have been more
specific."

pybites commented:
OK this one was really nasty. We have what we think is a workaround. 

Please test it next time you want to submit code and let us know: 

https://github.com/pybites/challenges/blob/master/INSTALL.md

Cheers
